src/SFA.DAS.ProviderCommitments/SFA.DAS.Commitments.Shared/Services/ModelMapper.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Microsoft.Extensions.DependencyInjection;
+using SFA.DAS.Commitments.Shared.Interfaces;
+
+namespace SFA.DAS.Commitments.Shared.Services
+{
+    public class ModelMapper : IModelMapper
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public ModelMapper(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public T Map<T>(object source) where T : class
+        {
+            var sourceType = source.GetType();
+            var destinationType = typeof(T);
+
+            Type[] typeArgs = { sourceType, destinationType };
+            var mapperType = typeof(IMapper<,>).MakeGenericType(typeArgs);
+
+            object mapper;
+
+            try
+            {
+                mapper = _serviceProvider.GetRequiredService(mapperType);
```

Comment:
Instead of catching an exception you could use the `GetService<T>` extension method and check for null.

Reply:
✔️